test/unit/specs/components/Personalization/createApplyPropositions.spec.js

```diff
@@ -0,0 +1,283 @@
+  {
+    id: "AT:eyJhY3Rpdml0eUlkIjoiNDQyMzU4IiwiZXhwZXJpZW5jZUlkIjoiIn0=",
+    scope: "home",
+    items: [
+      {
+        id: "442358",
+        schema: "https://ns.adobe.com/personalization/dom-action",
+        data: {
+          type: "click",
+          format: "application/vnd.adobe.target.dom-action",
+          selector: "#root"
+        }
+      }
+    ]
+  },
+  {
+    id: "AT:eyJhY3Rpdml0eUlkIjoiNDQyMzU4IiwiZXhwZXJpZW5jZUlkIjoiIn1=",
+    scope: "home",
+    items: [
+      {
+        id: "442359",
+        schema: "https://ns.adobe.com/personalization/html-content-item",
+        data: {
+          content: "<p>Some custom content for the home page</p>",
+          format: "text/html",
+          id: "1202448"
+        }
+      }
+    ]
+  },
+  {
+    id: "AT:eyJhY3Rpdml0eUlkIjoiNDQyMzU4IiwiZXhwZXJpZW5jZUlkIjoiIn1=",
+    scope: "home",
+    items: [
+      {
+        id: "442360",
+        schema: "https://ns.adobe.com/personalization/json-content-item",
+        data: {
+          content: "{'field1': 'custom content'}",
+          format: "text/javascript",
+          id: "1202449"
+        }
+      }
+    ],
+    renderAttempted: false
+  },
+  {
+    id: "AT:eyJhY3Rpdml0eUlkIjoiMTQxNjY0IiwiZXhwZXJpZW5jZUlkIjoiMCJ9",
+    scope: "home",
+    items: [
+      {
+        id: "xcore:personalized-offer:134ce877e13a04ca",
+        etag: "4",
+        schema:
+          "https://ns.adobe.com/experience/offer-management/content-component-html",
+        data: {
+          id: "xcore:personalized-offer:134ce877e13a04ca",
+          format: "text/html",
+          language: ["en-us"],
+          content: "<p>An html offer from Offer Decisioning</p>",
+          characteristics: {
+            testing: "true"
+          }
+        }
+      }
+    ]
+  },
+  {
+    id: "AT:eyJhY3Rpdml0eUlkIjoiNDQyMzU4IiwiZXhwZXJpZW5jZUlkIjoiIn1=",
+    scope: "__view__",
+    renderAttempted: true,
+    items: [
+      {
+        id: "442358",
+        schema: "https://ns.adobe.com/personalization/dom-action",
+        data: {
+          type: "click",
+          format: "application/vnd.adobe.target.dom-action",
+          selector: "#root"
+        }
+      }
+    ]
+  }
+];
+
+const METADATA = {
+  home: {
+    selector: "#home-item1",
+    actionType: "setHtml"
+  }
+};
+
+describe("Personalization::createApplyPropositions", () => {
+  let executeDecisions;
+
+  beforeEach(() => {
+    executeDecisions = jasmine.createSpy("executeDecisions");
+  });
+
+  it("it should return an empty propositions promise if propositions is empty array", () => {
+    executeDecisions.and.returnValue(
+      Promise.resolve(PAGE_WIDE_SCOPE_DECISIONS)
+    );
+
+    const applyPropositions = createApplyPropositions({
+      executeDecisions
+    });
+
+    return applyPropositions({
+      propositions: []
+    }).then(result => {
+      expect(result).toEqual({ propositions: [] });
+      expect(executeDecisions).toHaveBeenCalledTimes(1);
+    });
```

Comment:
In my opinion if `propositions` is an empty array we should short-circuit and the code wouldn't reach `executeDecisions`.

Reply:
That is my preference as well.  @jonsnyder, what do you think?

Reply:
I don't think it matters either way. You could remove the check that executeDecisions was called once.

Reply:
I would still recommend that we avoid calling `executeDecisions` in case the propositions array is empty to be less error prone.